tests/phpunit/tests/modules/modules.php

```diff
@@ -0,0 +1,410 @@
+<?php
+/**
+ * @group modules
+ * @covers ::wp_register_module
+ * @covers ::wp_enqueue_module
+ * @covers ::wp_dequeue_module
+ * @covers WP_Modules::print_enqueued_modules
+ * @covers WP_Modules::print_import_map
+ * @covers WP_Modules::print_module_preloads
+ */
+class Tests_Modules_Functions extends WP_UnitTestCase {
+	/**
+	 * Stores the original value of WP_Modules::$registered to restore it later.
+	 *
+	 * @var array
+	 */
+	protected $old_registered;
+
+	/**
+	 * Stores the original value of WP_Modules::$enqueued_before_registered to
+	 * restore it later.
+	 *
+	 * @var array
+	 */
+	protected $old_enqueued_before_registered;
+
+	public function set_up() {
+		parent::set_up();
+
+		$registered = new ReflectionProperty( 'WP_Modules', 'registered' );
+		$registered->setAccessible( true );
+		$this->old_registered = $registered->getValue();
+		$registered->setValue( null, array() );
+
+		$enqueued_before_registered = new ReflectionProperty( 'WP_Modules', 'enqueued_before_registered' );
+		$enqueued_before_registered->setAccessible( true );
+		$this->old_enqueued_before_registered = $enqueued_before_registered->getValue();
+		$enqueued_before_registered->setValue( null, array() );
+	}
+
+	public function tear_down() {
+		$registered = new ReflectionProperty( 'WP_Modules', 'registered' );
+		$registered->setAccessible( true );
+		$registered->setValue( null, $this->old_registered );
+
+		$enqueued_before_registered = new ReflectionProperty( 'WP_Modules', 'enqueued_before_registered' );
+		$enqueued_before_registered->setAccessible( true );
+		$enqueued_before_registered->setValue( null, $this->old_enqueued_before_registered );
+
+		parent::tear_down();
+	}
```

Comment:
See above, by using instances instead of a static class, this entire hack/method could be removed.

Reply:
Done in https://github.com/WordPress/wordpress-develop/pull/5818/commits/1aa45a6198e73869dcf4e642c3d2978888af612d.